datafusion-examples/examples/external_dependency/query-aws-s3.rs

```diff
@@ -63,5 +63,14 @@ async fn main() -> Result<()> {
     // print the results
     df.show().await?;
 
+    // dynamic query by the file path
+    ctx.enable_url_table();
+    let df = ctx
+        .sql(format!(r#"SELECT * FROM '{}' LIMIT 10"#, &path).as_str())
+        .await?;
+
+    // print the results
+    df.show().await?;
```

Comment:
Thanks, @alamb. I added a simple s3 example here. I hope it is what you want or that it could inspire you for a new example.

Reply:
No actually this is perfect -- thank you @goldmedal